rusk/src/bin/main.rs

```diff
@@ -74,9 +74,13 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
     tracing::subscriber::set_global_default(subscriber)?;
 
     let router = {
-        let rusk = Rusk::builder(disk_backend)
-            .path(rusk_profile::get_rusk_state_id_path()?)
-            .build()?;
+        let mut rusk_builder = Rusk::builder(disk_backend);
+
+        if let Some(key) = config.wallet.generator() {
+            rusk_builder = rusk_builder.generator(key);
+        }
```

Comment:
This is a bit ugly. I would suggest to change the builder API differently so that we can have the code as we did before:

```rust
let rusk = Rusk::builder(disk_backend)
    .path(rusk_profile::get_rusk_state_id_path()?)
    .generator(config.wallet.generator())
    .build()?;
````

I will add the comment how I think it should be the implementation of the API.